src/scripts/main.js

```diff
@@ -1,3 +1,56 @@
 'use strict';
 
-// write your code here
+const baseUrl
+= 'https://mate-academy.github.io/phone-catalogue-static/api/phones.json';
+const detailsUrl
+= 'https://mate-academy.github.io/phone-catalogue-static/api/phones/';
+
+const getPhones = (url) => {
+  fetch(url)
+    .then(response => {
+      if (!response.ok) {
+        return Promise.reject(new Error(
+          `${response.status} - ${response.statusText}`
+        ));
+      }
+    });
+
+  return new Promise(function(resolve, reject) {
+    fetch(url)
+      .then(response => resolve(response.json()));
+
+    setTimeout(() => {
+      reject(new Error('Error!'));
+    }, 5000);
+  });
+};
+
+const getPhonesDetails = (list) => {
+  const listOfPhones = list.map(phone => phone.id);
+
+  const preparedList = listOfPhones.map(
+    phoneId => fetch(
+      `${detailsUrl}${phoneId}.json`)
```

Comment:
форматування рядків 32-33

```suggestion
      const preparedList = listOfPhones.map(
        phoneId => fetch(`${detailsUrl}${phoneId}.json`)
          .then(response => response.json()
      )
```